### ISSUE TITLE
What's wrong about the 'Gradient overflow' when I using EfficientNet-b7?

### ISSUE THREAD

**Issue by Liam-Yang** [lukemelas/EfficientNet-PyTorch#65]
I have confirmed that the gradient has been zeroed before each training.
I used the EfficientNet-b7 on the Kaggle's kernel. My dataset includes about 30,000 images, and each image will be resized to 256*256 before training.
I used EfficientNet-b5 to run successfully,  then I replaced EfficientNet-b5 with EfficientNet-b7 and changed the batch-size from 32 to 8.
Then there appeared ‘Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0’， ‘Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0’，etc.

**Comment by bigbrother33** [comment]
did you use nvidia apex? it is no problem for that

**Comment by Liam-Yang** [comment]
> did you use nvidia apex? it is no problem for that

Yes, I  have installed the APEX tool. But the result is as I said above. I am very depressed.

**Comment by Luonic** [comment]
@Liam-Yang you shoudl read about how APEX implements adaptive loss scaling for accurate fp16 training. What you described above is absolutely normal and intended for APEX.

**Comment by aditya1709** [comment]
Have you been able to successfully train efficientNet using FP16?
